modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/inputs/DateTimeInput.es.js

```diff
@@ -11,35 +11,76 @@ class DateTimeInput extends React.Component {
 		value: propTypes.string
 	};
 
+	constructor(props) {
+		super(props);
+
+		const date = new Date(props.value);
+
+		this.state = {
+			value: dateFns.format(date, INPUT_DATE_FORMAT)
+		};
+	}
+
+	shouldComponentUpdate(nextProps) {
```

Comment:
You can use https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops to get around having to do this logic in both the constructor and shouldComponentUpdate